lib/core/source/common/http_service.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:core';
+
+import 'package:dio/dio.dart';
+import 'package:flutter_template/core/common/network_exceptions.dart';
+
+abstract class HttpService {
+  Future<Response> get(
+    String uri, {
+    Map<String, dynamic>? queryParameters,
+    Options? options,
+    CancelToken? cancelToken,
+    ProgressCallback? onReceiveProgress,
+  });
+
+  Future<Response> post(
+    String uri, {
+    data,
+    Map<String, dynamic>? queryParameters,
+    Options options,
+    CancelToken? cancelToken,
+    ProgressCallback? onSendProgress,
+    ProgressCallback? onReceiveProgress,
+  });
+
+  Future<Response> put(
+    String uri, {
+    data,
+    Map<String, dynamic>? queryParameters,
+    Options options,
+    CancelToken? cancelToken,
+    ProgressCallback? onSendProgress,
+    ProgressCallback? onReceiveProgress,
+  });
+
+  Future<Response> delete(
+    String uri, {
+    data,
+    Map<String, dynamic>? queryParameters,
+    Options options,
+    CancelToken? cancelToken,
+    ProgressCallback? onSendProgress,
+    ProgressCallback? onReceiveProgress,
+  });
+}
+
+// Sample api https://www.appsloveworld.com/sample-rest-api-url-for-testing-with-authentication
+class HttpServiceDio implements HttpService {
+  late Dio _dio;
+
+  HttpServiceDio() {
+    final options =
+        BaseOptions(baseUrl: 'http://restapi.adequateshop.com/api/');
+    _dio = Dio(options);
+    // dio.interceptors.add(DioCacheManager(
+    //   CacheConfig(
+    //     baseUrl: baseUrl,
+    //   ),
+    // ).interceptor);
+    // dio.interceptors.add(AuthInterceptor());
```

Comment:
Why this commented?

Reply:
we should do that in the future, but I think we should remove this comment. Then we can include it